addons/website_slides_sale/models/sale_order.py

```diff
@@ -0,0 +1,24 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models, api
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    @api.multi
+    def _action_confirm(self):
+        """ If the product of an order line is a 'course', we add the client of the sale_order
+        as a member of the channel(s) on which this product is configured (see slide.channel.product_id). """
+        result = super(SaleOrder, self)._action_confirm()
+
+        for sale_order in self:
```

Comment:
I think we could improve a bit computation here to try to speedup things. Indeed we have a mapped and a search inside a loop which will scale badly if recordset grows.

Maybe we could try doing a two-step search 
1/ search all products related to courses
2/ search so lines with so_id in self.ids, product_id in products.ids
Then add member accordingly.

Reply:
I did everything we talked about except the "list comprehension" since i use the "|" operator and it does not work (I think?) with a list.

Anyway it should be a lot quicker now.
I also modified the test a little to check for multiple channels configured with the same product.

Let me know if I still need to change anything.